### PR TITLE
HM4 

### PR DESCRIPTION
**Pull request by DadizhaO**

### CODE REVIEW

File: src/main/java/HM04/Gold.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package HM04;
+
+public class Gold implements BasicInterface {
+
+    private int quantity;
```
Comment (SerhiiPasko) [inline src/main/java/HM04/Gold.java:5]: You have this field and getter and same constructor in all implemantations. In this case good reason use abstract class with this logic

File: src/main/java/HM04/Bank.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package HM04;
+
+
+class Bank <T extends BasicInterface> {
+
```
Comment (SerhiiPasko) [inline src/main/java/HM04/Bank.java:5]: Main reason use generic on class level if you have fields of this generic type. For example -private T;

File: src/main/java/HM04/Main.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package HM04;
+
+public class Main {
+
+    public static void main(String[] args) {
+        Currency currency = new Currency(120);
+        Gold gold = new Gold(150);
+        Securities securities = new Securities(110);
+
+        Bank<BasicInterface> bank = new Bank<>();
```
Comment (SerhiiPasko) [inline src/main/java/HM04/Main.java:10]: You use Bank\<T\> as util class. It isn't reasons to use Bank class as generic

**Comment by Yanetta** [comment]
good

**Pull request merged**